### PR TITLE
Version 6.0.0 → 7.0.0

### PR DESCRIPTION
**Pull request by f-f**
I think it's time for another release:
- it's been almost two months since `6.0.0`
- we have quite many unreleased notable changes/additions (e.g. the empty alternatives)
- the [new release of `dhall-kubernetes`](https://github.com/dhall-lang/dhall-kubernetes/pull/57) is blocked by a new release of `dhall-haskell` (specifically `dhall-json`), which is blocked by a new release here
- I'd like to have a release before #494, as that's another big change that might take a while and there's no point in hurrying it

Checklist:
- wait for the following to be merged:
  - [x] #497 
  - [x] #495 
  - [x] #490 
  - [x] #475?
- [x] compile the `CHANGELOG`

### CODE REVIEW

**Comment by ari-becker** [comment]
I'd like to humbly request that (remind about?) [#396](https://github.com/dhall-lang/dhall-lang/issues/396) be fixed before cutting a new release. It has been open for nearly two months (since 5.0.0, when it was decided not to delay 6.0.0 for a fix) and affects [documented functionality](http://hackage.haskell.org/package/dhall-1.21.0/docs/Dhall-Tutorial.html#g:15) that is currently unusable/broken.

**Comment by Gabriel439** [comment]
@ari-becker: I prefer not to hold up releases for features that aren't close to complete.  We can always cut a new release whenever the fix for that is ready, since cutting new releases is cheap.

**Comment by Gabriel439** [comment]
@f-f: I can help write up the `CHANGELOG` if you don't mind

**Comment by f-f** [comment]
@Gabriel439 I was AFK for some days and I was just starting to look at the list of commits in order to write it. However I won't get around to work on it before tomorrow morning (EEST time), so feel free to push to this branch in the meanwhile 🙂

**Comment by f-f** [comment]
This is now ready for review, I'd say we couold wait three more days from now in order to review it and eventually wait for fixes coming from bringing implementations to parity with the standard

**Comment by Gabriel439** [comment]
The Haskell implementation will be ready once https://github.com/dhall-lang/dhall-haskell/pull/903 is merged tomorrow

**Comment by f-f** [comment]
@Gabriel439 thanks! 🙂 

I was thinking that we should define some kind of PR-freeze for releases. E.g. should we hold #504 and #507 until we merge this or should we wait for them? Because the Haskell implementation will need more adjustment to be on par. On the other hand they are small fixes that would be nice to get into this release

**Comment by Nadrieril** [comment]
Is it a problem if the Haskell implementation doesn't support the latest standard right away ?

**Comment by f-f** [comment]
@Nadrieril it's not a hard requirement, but I think having at least one implementation being "bug free" (i.e. with the latest standard and tests) at the time of a release ensures we're not cutting a release with broken tests and/or unspecified standard - this is because we don't have CI on the tests here, so we have to wait until at least one implementation takes them into use to be sure they are actually standard compliant

**Comment by Gabriel439** [comment]
@f-f: My general rule of thumb for release process is "trains run on time", meaning that whenever the decision is not clear, bias towards cutting a release earlier and delaying any changes in flight until the next release.  If releases are sufficiently frequent then there is no harm in delaying a change to the next release.

The thing that I want to avoid is people rushing to merge in changes before a release boundary out of fear of delaying until the next release, because that leads to poor quality in my experience.

**Comment by philandstuff** [comment]
While I would personally like to get #504 into this release, I think it’s also worth remembering that the standard is the source of truth, not the tests, so technically #504 doesn’t change what’s required of implementations. (Correspondingly, if implementations work to a hypothetical 7.0.0-plus-#504 standard, then they are necessarily also 7.0.0-compliant.)

In short: don’t wait for #504. If implementations want it, they can cherry-pick it in.

**Comment by philandstuff** [comment]
(Also we are allowed to bump non-major versions, so we could easily cut a 7.0.1 or 7.1.0 with #504 added)

**Comment by Gabriel439** [comment]
Yeah, that's correct.  We can always cut a minor release if necessary if an implementation wants to hew only to published releases

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,212 @@ file.
...
+    
+    For example this is now possible:
+    
+    ```hs
+    let Role = < Wizard | Fighter | Rogue >
+
+    let show : Role → Text
+        show =
+            λ(x : Role)
+          → merge { Wizard = "Wizard", Fighter = "Fighter", Rogue = "Rogue" } x
```
[the remark this answers is not in the document]
Comment (f-f) [reply CHANGELOG.md]: Good catch! Added 🙂

**Comment by f-f** [comment]
@Gabriel439 @philandstuff agreed 👍

Then I'd say we merge this now and cut the release, and I'll put together a PR for the releasing procedure to integrate the comments above plus detailing something like a freeze window.

**Pull request merged**